### ISSUE TITLE
module name collision

### ISSUE THREAD

**Issue by aligon** [rollup/rollup#1910]
https://gist.github.com/aligon/bba460bf9dadee2d37f2214ca854e77a

I'm using an external component from another library, that I want to pass the same props to from a couple of different places. To that end, I created a wrapper component that imports the external component and renders it with the shared props. The wrapper and external component share the same class name, so in the wrapper I import the external to another name. The bundled version of the wrapper's render appears to be rendering the wrapper component, not the external component.

/cc @jsg2021

**Comment by guybedford** [comment]
Is there any way you can share this either as a repl test case?

Something like - https://rollupjs.org/repl?version=0.55.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMFJlYWN0JTIwZnJvbSUyMCdyZWFjdCclM0IlNUNuJTJGJTJGJTIwaW1wb3J0JTIwUHJvcFR5cGVzJTIwZnJvbSUyMCdwcm9wLXR5cGVzJyUzQiU1Q25pbXBvcnQlMjAlN0JTdGlja3klMjBhcyUyMFJlYWN0U3RpY2t5JTdEJTIwZnJvbSUyMCcuJTJGcmVhY3Qtc3RpY2t5LmpzJyUzQiU1Q24lNUNuZXhwb3J0JTIwZGVmYXVsdCUyMGNsYXNzJTIwU3RpY2t5JTIwZXh0ZW5kcyUyMFJlYWN0LkNvbXBvbmVudCUyMCU3QiU1Q24lNUNuJTVDdHJlbmRlciUyMCgpJTIwJTdCJTVDbiU1Q3QlNUN0cmV0dXJuJTIwUmVhY3RTdGlja3koKSUzQiU1Q24lNUN0JTdEJTVDbiU3RCUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJyZWFjdC1zdGlja3kuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZnVuY3Rpb24lMjBTdGlja3klMjAoKSUyMCU3QiU1Q24lNUN0cmV0dXJuJTIwJ3JlYWN0LXN0aWNreSclM0IlNUNuJTdEJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

**Comment by jsg2021** [comment]
The repl page isn't generating a shareable url at the moment... here is a repro snippet...just pase this into the repl as the main module, and select es output...
```es
import { Sticky as ReactSticky } from 'react-sticky';

var Sticky = function () {
	function Sticky() {}

	Sticky.foo = ReactSticky;

	return Sticky;
}();

export { Sticky as default };
```

The original babel output was much longer, (and still repro'd) so I trimed it to the essential bits... it looks stupid now, but it reproduces the issue... ReactSticky is replaced with Sticky...which is incorrect.

**Comment by guybedford** [comment]
@jsg2021 thanks for the replication, that works out for me yeah -

https://rollupjs.org/repl?version=0.55.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMFN0aWNreSUyMGFzJTIwUmVhY3RTdGlja3klMjAlN0QlMjBmcm9tJTIwJ3JlYWN0LXN0aWNreSclM0IlNUNuJTVDbnZhciUyMFN0aWNreSUyMCUzRCUyMGZ1bmN0aW9uJTIwKCklMjAlN0IlNUNuJTVDdGZ1bmN0aW9uJTIwU3RpY2t5KCklMjAlN0IlN0QlNUNuJTVDbiU1Q3RTdGlja3kuZm9vJTIwJTNEJTIwUmVhY3RTdGlja3klM0IlNUNuJTVDbiU1Q3RyZXR1cm4lMjBTdGlja3klM0IlNUNuJTdEKCklM0IlNUNuJTVDbmV4cG9ydCUyMCU3QiUyMFN0aWNreSUyMGFzJTIwZGVmYXVsdCUyMCU3RCUzQiUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtYXRocy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjIlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

Looking into it now.

**Comment by guybedford** [comment]
I've posted a PR for the fix, should land soon hopefully. Thanks for isolating this one.

**Comment by bdwain** [comment]
Not sure if these are related or not but it seems similar to #1913 (except that the REPL seems to be working correctly now for this example, and not #1913)

**Comment by guybedford** [comment]
Fixed in 0.55.1.

**Comment by guybedford** [comment]
(Note the repl still needs updating)
